File: fix_const.py
import re
import os

log_path = r"C:\Users\ivann\.gemini\antigravity-ide\brain\8ecf8ca6-a4dd-480e-ba50-02dbde0a6ed5\.system_generated\tasks\task-79.log"
# Windows path
workspace_dir = r"c:\project flutter\stokwarung"

with open(log_path, 'r', encoding='utf-8') as f:
    lines = f.readlines()

fixed_files = set()

for line in lines:
    if 'invalid_constant' in line:
        match = re.search(r' - (lib\\[^:]+):(\d+):\d+ - ', line)
        if match:
            rel_file_path = match.group(1)
            file_path = os.path.join(workspace_dir, rel_file_path)
            line_num = int(match.group(2)) - 1
            
            try:
                with open(file_path, 'r', encoding='utf-8') as f_in:
                    file_lines = f_in.readlines()
                
                original = file_lines[line_num]
                # Try to remove the first 'const ' we find on that line
                file_lines[line_num] = re.sub(r'\bconst\s+', '', original, count=1)
                
                with open(file_path, 'w', encoding='utf-8') as f_out:
                    f_out.writelines(file_lines)
                print(f"Fixed {file_path}:{line_num+1}")
                fixed_files.add(file_path)
            except Exception as e:
                print(f"Error fixing {file_path}:{line_num+1} - {e}")

File: fix_const.dart
// ignore_for_file: avoid_print
import 'dart:io';

void main() {
  final logFile = File(r"C:\Users\ivann\.gemini\antigravity-ide\brain\8ecf8ca6-a4dd-480e-ba50-02dbde0a6ed5\.system_generated\tasks\task-122.log");
  final workspaceDir = r"c:\project flutter\stokwarung";
  for (var line in logFile.readAsLinesSync()) {
    if (line.contains('invalid_constant')) {
      final parts = line.split(' - ');
      if (parts.length >= 3) {
        final pathAndLine = parts[2].trim();
        final subParts = pathAndLine.split(':');
        if (subParts.length >= 3) {
          final relPath = subParts[0];
          final lineNum = int.parse(subParts[1]) - 1;
          final file = File('$workspaceDir\\$relPath');
          if (file.existsSync()) {
            final lines = file.readAsLinesSync();
            if (lineNum < lines.length) {
              final original = lines[lineNum];
              lines[lineNum] = original.replaceFirst(RegExp(r'\bconst\s+'), '');
              if (original != lines[lineNum]) {
                file.writeAsStringSync(lines.join('\n'));
                print('Fixed $relPath:$lineNum');
              }
            }
          }
        }
      }
    }
  }
}
